addons/mail/tests/test_mail_template.py

```diff
@@ -70,12 +70,57 @@ def test_mail_template_acl(self):
         with self.assertRaises(AccessError):
             self.env['mail.template'].with_user(self.user_employee).create({'body_html': '<p t-esc="\'foo\'"></p>'})
 
+        # Standard employee cannot edit templates from another user, non-dynamic and dynamic
+        with self.assertRaises(AccessError):
+            mail_template.with_user(self.user_employee).body_html = '<p>foo</p>'
         with self.assertRaises(AccessError):
             mail_template.with_user(self.user_employee).body_html = '<p t-esc="\'foo\'"></p>'
 
+        # Standard employee can edit his own templates if not dynamic
+        employee_template.with_user(self.user_employee).body_html = '<p>foo</p>'
+
         # Standard employee cannot create and edit templates with dynamic inline fields
         with self.assertRaises(AccessError):
             self.env['mail.template'].with_user(self.user_employee).create({'email_to': '{{ object.partner_id.email }}'})
 
+        # Standard employee cannot edit his own templates if dynamic
+        with self.assertRaises(AccessError):
+            employee_template.with_user(self.user_employee).body_html = '<p t-esc="\'foo\'"></p>'
+
+        with self.assertRaises(AccessError):
+            employee_template.with_user(self.user_employee).email_to = '{{ object.partner_id.email }}'
+
+    def test_mail_template_acl_translation(self):
+        ''' Test that a user that doenn't have the group_mail_template_editor cannot create / edit
+        translation with dynamic code if he cannot write dynamic code on the related record itself.
+        '''
+
+        self.env.ref('base.lang_fr').sudo().active = True
+
+        employee_template = self.env['mail.template'].with_user(self.user_employee).create({'body_html': '<p>foo</p>'})
```

Comment:
Could you set a model (to have a coherent template) ? And maybe a subject so that we also that we try inline translated fields (in addition to qweb).

Reply:
I can add model and co, but same for all previous tests in this case :thinking: ... 

About the inline, I can add a test again if you really want, but it is already tested by previous test `test_mail_template_acl`.
This one, just test that the _is_dynamic call is done on translation create/write too.

Reply:
Well indeed, maybe there are other instances but always better to at least give a model. The field is not required, but it may become required someday (as it makes no sense to have generic templates), better be already compliant :) .

For subject, indeed it is somehow already tested (at mixin / template level), but as you test the manual write on the translation for body, we could also test the manual write on translation for subject. That way we test the mixin, and the translations, at both levels.

Reply:
done